libcaf_core/caf/logger.hpp

```diff
@@ -353,8 +353,10 @@ class CAF_CORE_EXPORT logger {
 #define CAF_PUSH_AID(aarg)                                                     \
   caf::actor_id CAF_PP_UNIFYN(caf_aid_tmp)                                     \
     = caf::logger::thread_local_aid(aarg);                                     \
-  auto CAF_PP_UNIFYN(caf_aid_tmp_guard) = caf::detail::make_scope_guard(       \
-    [=] { caf::logger::thread_local_aid(CAF_PP_UNIFYN(caf_aid_tmp)); })
+  auto CAF_PP_UNIFYN(caf_aid_tmp_guard)                                        \
+    = caf::detail::scope_guard([=]() noexcept {                                \
+        caf::logger::thread_local_aid(CAF_PP_UNIFYN(caf_aid_tmp));             \
+      })
```

Comment:
If we declare this lambda `noexcept`, we'll also need to declare `thread_local_aid` `noexcept`.

Reply:
I will add it to logger. Just a note here, I tried following the chain and adding `noexcept` everywhere and failed. As you noted in the previous comment as well, there are places where it's just not possible.